hosted-cluster-config-operator/controllers/resources/crd/reconcile.go

```diff
@@ -0,0 +1,38 @@
+package crd
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/openshift/hypershift/hosted-cluster-config-operator/api"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	//go:embed assets/apiserver.openshift.io_apirequestcount-crd.yaml
+	requestCountCRDBytes []byte
+
+	requestCountCRD = mustCRD(requestCountCRDBytes)
+)
+
+func ReconcileRequestCountCRD(crd *apiextensionsv1.CustomResourceDefinition) error {
+	crd.Spec = requestCountCRD.Spec
+	return nil
+}
+
+func mustCRD(content []byte) *apiextensionsv1.CustomResourceDefinition {
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	deserializeResource(content, crd)
+	return crd
+}
+
+func deserializeResource(data []byte, obj runtime.Object) {
```

Comment:
What is the advantage of this over a simple `yaml.Unmarshal`?

Reply:
as far as I can tell, this uses the json serializer package in apimachinery which sets up some config defaults for json-iter along with a custom number parser: https://github.com/kubernetes/apimachinery/blob/release-1.22/pkg/runtime/serializer/json/json.go#L152-L159